haaska.py

```diff
@@ -452,6 +452,14 @@ def turn_on(self):
     def turn_off(self):
         self._call_service('cover/close_cover')
 
+    def set_percentage(self, val):
+        self._call_service('cover/set_cover_position', {'position': val})
+
+    def get_percentage(self):
+        state = self.ha.get('states/' + self.entity_id)
+        value = int(state['current_position'])
```

Comment:
Not all covers report their current_position (specifically Somfy zwave blinds). I'm doing a terrible hack to make this work on my system:

```python
    def get_percentage(self):
        return 50

    def set_percentage(self, val):
        if val > 50:
            self._call_service('cover/open_cover')
        if val < 50:
            self._call_service('cover/close_cover')
```

Definitely not saying to do that but you need to check if `current_position` is None to be safe here.